### PR TITLE
Text rendering

### PR DESCRIPTION
**Pull request by Joerseven**
Added text rendering. The relevant additions are:

A Font class which stores information about fonts. Examples of unique and shared pointers inside to look at.
The previous debug text method has been replaced by this text rendering method.
How you would use this interface includes:

1. Creating a font using the "LoadFont(_font path (no prefix)_). And keep a pointer to this font.
2. Use the RenderText function in Renderer to be called every frame. Debug text automatically does this so the debug can be used as usual.

Look in the constructor for an example of how the font is loaded and the render text for an example of how the font is drawn every frame. 

The font class doesn't contain any rendering specific font information, this is why the load and draw font functions are within the renderer which is specific to opengl.

### CODE REVIEW

File: Client/Font.h [hunk 1]
```diff
@@ -16,15 +20,17 @@ class Font {
     };
 
     struct Character {
-        unsigned int TextureID;  // ID handle of the glyph texture
-        Vector2i   Size;       // Size of glyph
-        Vector2i  Bearing;    // Offset from baseline to left/top of glyph
-        unsigned int Advance;    // Offset to advance to next glyph
+        unsigned int textureId;
```
Comment (Joerseven) [inline Client/Font.h:23]: what the hell why

File: Client/Renderer.h [hunk 2]
```diff
@@ -63,6 +65,12 @@ namespace NCL {
 			GLuint		shadowFBO;
 			Matrix4     shadowMatrix;
 
+            std::unique_ptr<Font> debugFont;
+            std::shared_ptr<OGLShader> textShader;
+
+            // Ortho for UI
+            Matrix4 uiOrthoView;
```
Comment (Joerseven) [inline Client/Renderer.h:72]: Change the name dumbass

**Review by Joerseven — COMMENTED** [review]
Lgtm!

**Review by DelCherub — APPROVED** [review]
Looks alright to me

**Pull request merged**